MetaMask/metamask-extension
# Display popup appStateController using ApprovalController

## Explanation

Changes the AppStateController to use ApprovalController instead of `showUserConfirmation` callback to display the popup.

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/405

## Manual Testing Steps

- Restart your extension so you have sure it's locked
- Go to [dapp test page](https://metamask.github.io/test-dapp/)
- Try to connect
- A popup will be displayed to unlock MM and connect

result: Operation should work as usual, connecting to the dapp test and enabling to perform any action there.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
